includes/integration.php

```diff
@@ -0,0 +1,537 @@
+            
+            add_action('wp_ajax_metrilo_import', array($this, 'metrilo_import'));
+            add_action('admin_menu', array($this, 'setup_admin_pages'));
+        }
+        
+        public function ensure_identify(){
+            // if user is logged in
+            if( !is_admin() && is_user_logged_in() && !( $this->session_get( $this->get_identify_cookie_name() ) ) ){
+                $user       = wp_get_current_user();
+                $user_roles = $user->roles;
+                $this->identify_call_data = $user->user_email;
+                
+                // check if roles should be sent and if they exist
+                if($this->send_roles_as_tags == 'yes' && !empty($user_roles)){
+                    $this->user_tags = $user_roles;
+                    $this->session_set($this->get_identify_cookie_name(), $user_roles);
+                }
+                $this->session_set($this->get_identify_cookie_name(), 'true');
+            }
+        }
+        
+        public function render_identify()
+        {
+            include_once(METRILO_ANALYTICS_PLUGIN_PATH . 'views/render-identify.php');
+        }
+        
+        public function render_events()
+        {
+            include_once(METRILO_ANALYTICS_PLUGIN_PATH . 'views/render-tracking-events.php');
+        }
+        
+        public function render_footer_events(){
+            include_once(METRILO_ANALYTICS_PLUGIN_PATH.'views/render-footer-tracking-events.php');
+        }
+        
+        public function woocommerce_footer_tracking(){
+            if($this->events_queue) {
+                $this->render_footer_events();
+            }
+        }
+        
+        public function render_snippet()
+        {
+            // check if we should track data for this user (if user is available)
+            if (!is_admin() && is_user_logged_in()) {
+                $user = wp_get_current_user();
+                if ($user->roles && $this->ignore_for_roles) {
+                    foreach ($user->roles as $r) {
+                        if (in_array($r, $this->ignore_for_roles)) {
+                            $this->accept_tracking = false;
+                        }
+                    }
+                }
+            }
+            
+            // render the JS tracking code
+            include_once(METRILO_ANALYTICS_PLUGIN_PATH . 'views/metrilo-front-end-library.php');
+        }
+        
+        public function woocommerce_tracking()
+        {
+            include_once(METRILO_ANALYTICS_PLUGIN_PATH . 'includes/tracking-events.php');
+        }
+        
+        public function add_to_cart(
+            $cart_item_key,
+            $product_id,
+            $quantity,
+            $variation_id = false,
+            $variation = false,
+            $cart_item_data = false
+        )
+        {
+            $product_id = ($variation_id) ? $variation_id : $product_id;
```

Comment:
Same as without the brackets:
```suggestion
            $product_id = $variation_id ? $variation_id : $product_id;
```

Reply:
Both are fixed.